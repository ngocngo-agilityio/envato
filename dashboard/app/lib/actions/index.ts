export * from './events';
export * from './pinCode';
export * from './transferMoney';
