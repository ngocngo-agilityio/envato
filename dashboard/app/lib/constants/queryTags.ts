export const QUERY_TAGS = {
  EVENTS: 'events',
  USERS: 'users',
};
