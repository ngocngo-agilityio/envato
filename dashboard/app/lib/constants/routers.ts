import { AUTHENTICATION_ROLE } from './authentication';

export const ROUTES = {
  ROOT: '/',
  LOGIN: 'login',
  REGISTER: 'register',
  TRANSACTION: 'transactions',
  MY_WALLET: 'my-wallets',
  USER: 'users',
  CALENDAR: 'calendar',
  HISTORY: 'histories',
  SETTING: 'settings',
  FORGOT_PASSWORD: 'forgot-password',
  SIGN_OUT: 'sign-out',
  NOT_FOUND: '*',
  SUPPORT: 'supports',
  INBOX: 'inbox',
  PRODUCT: 'products',
  RECENT_ACTIVITIES: 'recent-activities',
};

export const COMMON_ROUTES = [
  {
    id: 1,
    path: ROUTES.NOT_FOUND,
  },
];

export const PRIVATE_ROUTES = (role: string) => [
  {
    id: 1,
    path: ROUTES.ROOT,
  },
  {
    id: 2,
    path: ROUTES.TRANSACTION,
  },
  {
    id: 3,
    path: ROUTES.MY_WALLET,
  },
  {
    id: 4,
    path: ROUTES.HISTORY,
  },
  {
    id: 5,
    path: ROUTES.SETTING,
  },
  {
    ...(role === AUTHENTICATION_ROLE.SUPER_ADMIN && {
      id: 6,
      path: ROUTES.USER,
    }),
  },
  {
    id: 7,
    path: ROUTES.PRODUCT,
  },
  {
    id: 8,
    path: ROUTES.RECENT_ACTIVITIES,
  },
  {
    id: 9,
    path: ROUTES.CALENDAR,
  },
  {
    ...(role === AUTHENTICATION_ROLE.SUPER_ADMIN && {
      id: 10,
      path: ROUTES.INBOX,
    }),
  },
  {
    id: 11,
    path: ROUTES.SUPPORT,
  },
];

export const PUBLIC_ROUTES = [
  {
    id: 1,
    path: ROUTES.LOGIN,
  },
  {
    id: 2,
    path: ROUTES.REGISTER,
  },
  {
    id: 3,
    path: ROUTES.FORGOT_PASSWORD,
  },
  {
    id: 4,
    path: ROUTES.SIGN_OUT,
  },
];
