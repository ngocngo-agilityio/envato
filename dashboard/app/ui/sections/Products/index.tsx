'use client';
import { InView } from 'react-intersection-observer';
import { Box, Flex, Grid, GridItem } from '@chakra-ui/react';
import dynamic from 'next/dynamic';

// dynamic loading components
const ProductsTable = dynamic(() => import('@/ui/components/ProductsTable'));
const CardPayment = dynamic(() => import('@/ui/components/CardPayment'));
const BoxChat = dynamic(() => import('@/ui/components/BoxChat'));

const Products = () => (
  <Grid
    bg="background.body.primary"
    py={12}
    px={{ base: 6, xl: 12 }}
    templateColumns={{ base: 'repeat(1, 1fr)', '2xl': 'repeat(4, 1fr)' }}
    display={{ sm: 'block', md: 'grid' }}
    gap={{ base: 0, '2xl': 12 }}
  >
    <InView>
      {({ inView, ref }) => (
        <GridItem colSpan={3} ref={ref}>
          <Box
            as="section"
            bgColor="background.component.primary"
            borderRadius={8}
            px={6}
            py={5}
          >
            {inView && <ProductsTable />}
          </Box>
        </GridItem>
      )}
    </InView>
    <InView>
      {({ inView, ref }) => (
        <GridItem mt={{ base: 6, '2xl': 0 }} ref={ref}>
          <Flex direction={{ base: 'column', lg: 'row', xl: 'column' }} gap={6}>
            {inView && <CardPayment />}
            {inView && <BoxChat />}
          </Flex>
        </GridItem>
      )}
    </InView>
  </Grid>
);

export default Products;
