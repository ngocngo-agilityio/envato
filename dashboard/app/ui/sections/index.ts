export { default as UsersSection } from './Users';
export { default as SettingsSection } from './Settings';
export { default as SupportsSection } from './Support';
export { default as ForgotPasswordSection } from './Forgot';
export { default as AccountSection } from './Account';
export { default as DashBoardSection } from './DashBoard';
export { default as MyWalletSection } from './MyWallet';
export { default as TransactionSection } from './Transaction';
export { default as HistorySection } from './History';
export { default as ProductsSection } from './Products';
export { default as RecentActivitiesSection } from './RecentActivities';
export { default as Calendar } from './Calendar';
