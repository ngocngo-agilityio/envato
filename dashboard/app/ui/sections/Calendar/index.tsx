'use client';

// Libs
import { useCallback, useMemo, useState } from 'react';
import { Box, Grid, GridItem, useToast } from '@chakra-ui/react';
import { InView } from 'react-intersection-observer';
import dynamic from 'next/dynamic';
import dayjs from 'dayjs';

// Hooks
import { useEvents } from '@/lib/hooks';

// Types
import { TEvent } from '@/lib/interfaces';

// Constants
import { ERROR_MESSAGES, STATUS, SUCCESS_MESSAGES } from '@/lib/constants';

// Utils
import { customToast } from '@/lib/utils';

// Components
import { Calendar } from '@/ui/components';

// dynamic loading components
const CardPayment = dynamic(() => import('@/ui/components/CardPayment'));

const CalendarSection = () => {
  const toast = useToast();
  const [date, setDate] = useState(new Date());

  // Events
  const {
    data: events = [],
    isLoading: isLoadingEvents,
    isAddEvent,
    addEvent,
    isUpdateEvent,
    updateEvent,
    isDeleteEvent,
    deleteEvent,
  } = useEvents();

  const isLoading =
    isLoadingEvents || isAddEvent || isUpdateEvent || isDeleteEvent;

  const formattedEvents = useMemo(
    () =>
      events.map((event) => {
        const { eventName = '', startTime, endTime } = event || {};

        return {
          ...event,
          title: eventName,
          start: dayjs(startTime).toDate(),
          end: dayjs(endTime).toDate(),
        };
      }),
    [events],
  );

  const handleAddEventSuccess = useCallback(
    (eventDate: Date) => {
      setDate(eventDate);

      toast(
        customToast(
          SUCCESS_MESSAGES.CREATE_EVENT_SUCCESS.title,
          SUCCESS_MESSAGES.CREATE_EVENT_SUCCESS.description,
          STATUS.SUCCESS,
        ),
      );
    },
    [toast],
  );

  const handleAddEventError = useCallback(() => {
    toast(
      customToast(
        ERROR_MESSAGES.CREATE_EVENT_FAIL.title,
        ERROR_MESSAGES.CREATE_EVENT_FAIL.description,
        STATUS.ERROR,
      ),
    );
  }, [toast]);

  const handleAddEvent = useCallback(
    (data: Omit<TEvent, '_id'>) => {
      const { startTime } = data;
      const eventDate = new Date(startTime);

      addEvent(data, {
        onSuccess: () => handleAddEventSuccess(eventDate),
        onError: handleAddEventError,
      });
    },
    [addEvent, handleAddEventError, handleAddEventSuccess],
  );

  const handleUpdateEventSuccess = useCallback(
    (eventDate: Date) => {
      setDate(eventDate);

      toast(
        customToast(
          SUCCESS_MESSAGES.UPDATE_EVENT_SUCCESS.title,
          SUCCESS_MESSAGES.UPDATE_EVENT_SUCCESS.description,
          STATUS.SUCCESS,
        ),
      );
    },
    [toast],
  );

  const handleUpdateEventError = useCallback(() => {
    toast(
      customToast(
        ERROR_MESSAGES.UPDATE_EVENT_FAIL.title,
        ERROR_MESSAGES.UPDATE_EVENT_FAIL.description,
        STATUS.ERROR,
      ),
    );
  }, [toast]);

  const handleUpdateEvent = useCallback(
    (data: TEvent) => {
      const { startTime } = data;
      const eventDate = new Date(startTime);

      const payload = {
        ...data,
        eventId: data._id,
      };

      updateEvent(payload, {
        onSuccess: () => handleUpdateEventSuccess(eventDate),
        onError: handleUpdateEventError,
      });
    },
    [handleUpdateEventError, handleUpdateEventSuccess, updateEvent],
  );

  const handleDeleteEventSuccess = useCallback(() => {
    toast(
      customToast(
        SUCCESS_MESSAGES.DELETE_EVENT_SUCCESS.title,
        SUCCESS_MESSAGES.DELETE_EVENT_SUCCESS.description,
        STATUS.SUCCESS,
      ),
    );
  }, [toast]);

  const handleDeleteEventError = useCallback(() => {
    toast(
      customToast(
        ERROR_MESSAGES.DELETE_EVENT_FAIL.title,
        ERROR_MESSAGES.DELETE_EVENT_FAIL.description,
        STATUS.ERROR,
      ),
    );
  }, [toast]);

  const handleDeleteEvent = useCallback(
    (eventId: string) => {
      const payload = {
        eventId,
      };

      deleteEvent(payload, {
        onSuccess: handleDeleteEventSuccess,
        onError: handleDeleteEventError,
      });
    },
    [deleteEvent, handleDeleteEventError, handleDeleteEventSuccess],
  );

  return (
    <Grid
      bg="background.body.primary"
      py={12}
      px={{ base: 6, xl: 12 }}
      templateColumns={{ base: 'repeat(1, 1fr)', '2xl': 'repeat(4, 1fr)' }}
      display={{ sm: 'block', md: 'grid' }}
      gap={{ base: 0, '2xl': 12 }}
    >
      <GridItem colSpan={3}>
        <Box
          as="section"
          bgColor="background.component.primary"
          borderRadius={8}
          px={{ base: 4, md: 10 }}
          py={{ base: 4, md: 5 }}
        >
          <Calendar
            events={formattedEvents}
            date={date}
            isLoading={isLoading}
            onSetDate={setDate}
            onAddEvent={handleAddEvent}
            onEditEvent={handleUpdateEvent}
            onDeleteEvent={handleDeleteEvent}
          />
        </Box>
      </GridItem>

      <InView>
        {({ inView, ref }) => (
          <GridItem mt={{ base: 6, '2xl': 0 }} ref={ref}>
            {inView && <CardPayment />}
          </GridItem>
        )}
      </InView>
    </Grid>
  );
};

export default CalendarSection;
