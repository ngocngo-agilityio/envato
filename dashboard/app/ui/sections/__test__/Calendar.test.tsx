// Libs
import { screen } from '@testing-library/react';

// Utils
import { customToast, renderQueryProviderTest } from '@/lib/utils';

// Constants
import { ERROR_MESSAGES, STATUS, SUCCESS_MESSAGES } from '@/lib/constants';

// Mocks
import {
  MOCK_CALENDAR_NOW_DATE,
  MOCK_EVENTS,
  MOCK_UPDATE_EVENT_FORM,
} from '@/lib/mocks';

// Sections
import { Calendar } from '@/ui/sections';
import { useEvents } from '@/lib/hooks';

const mockAddEvent = jest.fn();
const mockUpdateEvent = jest.fn();
const mockDeleteEvent = jest.fn();

jest.mock('@/lib/hooks', () => ({
  ...jest.requireActual('@/lib/hooks'),
  useEvents: jest.fn(),
}));

jest.mock('@/lib/utils', () => ({
  ...jest.requireActual('@/lib/utils'),
  customToast: jest.fn(),
}));

describe('Calendar section', () => {
  beforeEach(() => {
    jest.useFakeTimers({ now: MOCK_CALENDAR_NOW_DATE });

    (useEvents as jest.MockedFunction<typeof Object>).mockReturnValue({
      isLoading: false,
      data: MOCK_EVENTS,
      isAddEvent: false,
      addEvent: mockAddEvent,
      isUpdateEvent: false,
      updateEvent: mockUpdateEvent,
      isDeleteEvent: false,
      deleteEvent: mockDeleteEvent,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render match with snapshot.', () => {
    const { container } = renderQueryProviderTest(<Calendar />);

    expect(container).toMatchSnapshot();
  });

  it('Should render Calendar successfully with no events.', async () => {
    (useEvents as jest.MockedFunction<typeof Object>).mockReturnValue({
      data: undefined,
    });

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    expect(screen.getByText('Calendar')).toBeInTheDocument();
  });

  it('Should render Calendar successfully with event is array null', async () => {
    (useEvents as jest.MockedFunction<typeof Object>).mockReturnValue({
      data: [null],
    });

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    expect(screen.getByText('Month')).toBeInTheDocument();
  });

  it('Should handle updating event successfully', async () => {
    mockUpdateEvent.mockImplementationOnce((_, { onSuccess }) => onSuccess());

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    // Click event to open the event details modal
    await act(async () => {
      fireEvent.click(screen.getByText(MOCK_EVENTS[0].eventName));
    });

    // Click Edit icon to open the Edit Event form modal
    await act(async () => {
      fireEvent.click(screen.getByTestId('edit-icon'));
    });

    // Update Event
    await act(async () => {
      fireEvent.change(screen.getByLabelText('Title'), {
        target: { value: MOCK_UPDATE_EVENT_FORM.title },
      });
      fireEvent.click(screen.getByRole('button', { name: 'Save' }));
    });

    await waitFor(() => {
      expect(customToast).toHaveBeenCalledWith(
        SUCCESS_MESSAGES.UPDATE_EVENT_SUCCESS.title,
        SUCCESS_MESSAGES.UPDATE_EVENT_SUCCESS.description,
        STATUS.SUCCESS,
      );
    });
  });

  it('Should handle updating event failed', async () => {
    mockUpdateEvent.mockImplementationOnce((_, { onError }) => onError());

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    // Click event to open the event details modal
    await act(async () => {
      fireEvent.click(screen.getByText(MOCK_EVENTS[0].eventName));
    });

    // Click Edit icon to open the Edit Event form modal
    await act(async () => {
      fireEvent.click(screen.getByTestId('edit-icon'));
    });

    // Update Event
    await act(async () => {
      fireEvent.change(screen.getByLabelText('Title'), {
        target: { value: MOCK_UPDATE_EVENT_FORM.title },
      });
      fireEvent.click(screen.getByRole('button', { name: 'Save' }));
    });

    await waitFor(() => {
      expect(customToast).toHaveBeenCalledWith(
        ERROR_MESSAGES.UPDATE_EVENT_FAIL.title,
        ERROR_MESSAGES.UPDATE_EVENT_FAIL.description,
        STATUS.ERROR,
      );
    });
  });

  it('Should delete event successfully', async () => {
    mockDeleteEvent.mockImplementationOnce((_, { onSuccess }) => onSuccess());

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    // Click event to open the event details modal
    await act(async () => {
      fireEvent.click(screen.getByText(MOCK_EVENTS[0].eventName));
    });

    // Click Delete icon to open the confirm delete modal
    await act(async () => {
      fireEvent.click(screen.getByTestId('delete-icon'));
    });

    // Click Delete button on the confirm delete modal
    await act(async () => {
      fireEvent.click(screen.getByRole('button', { name: 'Delete' }));
    });

    await waitFor(() => {
      expect(customToast).toHaveBeenCalledWith(
        SUCCESS_MESSAGES.DELETE_EVENT_SUCCESS.title,
        SUCCESS_MESSAGES.DELETE_EVENT_SUCCESS.description,
        STATUS.SUCCESS,
      );
    });
  });

  it('Should delete event failed', async () => {
    mockDeleteEvent.mockImplementationOnce((_, { onError }) => onError());

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    // Click event to open the event details modal
    await act(async () => {
      fireEvent.click(screen.getByText(MOCK_EVENTS[0].eventName));
    });

    // Click Delete icon to open the confirm delete modal
    await act(async () => {
      fireEvent.click(screen.getByTestId('delete-icon'));
    });

    // Click Delete button on the confirm delete modal
    await act(async () => {
      fireEvent.click(screen.getByRole('button', { name: 'Delete' }));
    });

    await waitFor(() => {
      expect(customToast).toHaveBeenCalledWith(
        ERROR_MESSAGES.DELETE_EVENT_FAIL.title,
        ERROR_MESSAGES.DELETE_EVENT_FAIL.description,
        STATUS.ERROR,
      );
    });
  });

  it('Should add event successfully', async () => {
    const mockEvents = MOCK_EVENTS;
    mockEvents[0]._id = '';
    (useEvents as jest.MockedFunction<typeof Object>).mockReturnValue({
      data: MOCK_EVENTS,
      addEvent: mockAddEvent,
    });

    mockAddEvent.mockImplementationOnce((_, { onSuccess }) => onSuccess());

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    await act(async () => {
      fireEvent.click(screen.getByText(MOCK_EVENTS[0].eventName));
    });

    await act(async () => {
      fireEvent.click(screen.getByTestId('edit-icon'));
    });

    await act(async () => {
      fireEvent.change(screen.getByLabelText('Title'), {
        target: { value: MOCK_UPDATE_EVENT_FORM.title },
      });
      fireEvent.click(screen.getByRole('button', { name: 'Save' }));
    });

    await waitFor(() => {
      expect(customToast).toHaveBeenCalledWith(
        SUCCESS_MESSAGES.CREATE_EVENT_SUCCESS.title,
        SUCCESS_MESSAGES.CREATE_EVENT_SUCCESS.description,
        STATUS.SUCCESS,
      );
    });
  });

  it('Should add event failed', async () => {
    const mockEvents = MOCK_EVENTS;
    mockEvents[0]._id = '';
    (useEvents as jest.MockedFunction<typeof Object>).mockReturnValue({
      data: MOCK_EVENTS,
      addEvent: mockAddEvent,
    });

    mockAddEvent.mockImplementationOnce((_, { onError }) => onError());

    await act(async () => {
      renderQueryProviderTest(<Calendar />);
    });

    await act(async () => {
      fireEvent.click(screen.getByText(MOCK_EVENTS[0].eventName));
    });

    await act(async () => {
      fireEvent.click(screen.getByTestId('edit-icon'));
    });

    await act(async () => {
      fireEvent.change(screen.getByLabelText('Title'), {
        target: { value: MOCK_UPDATE_EVENT_FORM.title },
      });
      fireEvent.click(screen.getByRole('button', { name: 'Save' }));
    });

    await waitFor(() => {
      expect(customToast).toHaveBeenCalledWith(
        ERROR_MESSAGES.CREATE_EVENT_FAIL.title,
        ERROR_MESSAGES.CREATE_EVENT_FAIL.description,
        STATUS.ERROR,
      );
    });
  });
});
