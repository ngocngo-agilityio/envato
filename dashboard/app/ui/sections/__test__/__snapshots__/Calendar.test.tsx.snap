// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar section Should render match with snapshot. 1`] = `
<div>
  <div
    class="css-1qfa3yr"
  >
    <div
      class="css-rklm6r"
    >
      <section
        class="css-e61c1n"
      >
        <div
          class="rbc-calendar"
        >
          <div
            class="css-l0rj33"
          >
            <h3
              class="chakra-heading css-djphl8"
            >
              Calendar
            </h3>
            <h3
              class="chakra-heading css-1szuvtm"
            >
              May 2024
            </h3>
            <div
              class="css-1119wa9"
            >
              <div
                class="css-12wezbe"
              >
                <button
                  class="chakra-button css-63sota"
                  type="button"
                >
                  Month
                </button>
                <button
                  class="chakra-button css-1bvnu45"
                  type="button"
                >
                  Week
                </button>
                <button
                  class="chakra-button css-1bvnu45"
                  type="button"
                >
                  Day
                </button>
              </div>
              <div
                class="css-1oz3cly"
              >
                <button
                  aria-label="btn-back"
                  class="chakra-button css-19ua7xs"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="30"
                    stroke="#1A202C"
                    style="transform: rotate(90deg);"
                    viewBox="0 0 24 24"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 10L12 14L17 10"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </button>
                <button
                  aria-label="btn-next"
                  class="chakra-button css-19ua7xs"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="30"
                    stroke="#1A202C"
                    style="transform: rotate(-90deg);"
                    viewBox="0 0 24 24"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 10L12 14L17 10"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            aria-label="Month View"
            class="rbc-month-view"
            role="table"
          >
            <div
              class="rbc-row rbc-month-header"
              role="row"
            >
              <div
                class="rbc-header"
              >
                <span
                  aria-sort="none"
                  role="columnheader"
                >
                  Sun
                </span>
              </div>
              <div
                class="rbc-header"
              >
                <span
                  aria-sort="none"
                  role="columnheader"
                >
                  Mon
                </span>
              </div>
              <div
                class="rbc-header"
              >
                <span
                  aria-sort="none"
                  role="columnheader"
                >
                  Tue
                </span>
              </div>
              <div
                class="rbc-header"
              >
                <span
                  aria-sort="none"
                  role="columnheader"
                >
                  Wed
                </span>
              </div>
              <div
                class="rbc-header"
              >
                <span
                  aria-sort="none"
                  role="columnheader"
                >
                  Thu
                </span>
              </div>
              <div
                class="rbc-header"
              >
                <span
                  aria-sort="none"
                  role="columnheader"
                >
                  Fri
                </span>
              </div>
              <div
                class="rbc-header"
              >
                <span
                  aria-sort="none"
                  role="columnheader"
                >
                  Sat
                </span>
              </div>
            </div>
            <div
              class="rbc-month-row"
              role="rowgroup"
            >
              <div
                class="rbc-row-bg"
              >
                <div
                  class="rbc-day-bg rbc-off-range-bg"
                />
                <div
                  class="rbc-day-bg rbc-off-range-bg"
                />
                <div
                  class="rbc-day-bg rbc-off-range-bg"
                />
                <div
                  class="rbc-day-bg rbc-today"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
              </div>
              <div
                class="rbc-row-content"
                role="row"
              >
                <div
                  class="rbc-row "
                >
                  <div
                    class="rbc-date-cell rbc-off-range"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      28
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell rbc-off-range"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      29
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell rbc-off-range"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      30
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell rbc-now rbc-current"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      01
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      02
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      03
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      04
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rbc-month-row"
              role="rowgroup"
            >
              <div
                class="rbc-row-bg"
              >
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
              </div>
              <div
                class="rbc-row-content"
                role="row"
              >
                <div
                  class="rbc-row "
                >
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      05
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      06
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      07
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      08
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      09
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      10
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      11
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rbc-month-row"
              role="rowgroup"
            >
              <div
                class="rbc-row-bg"
              >
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
              </div>
              <div
                class="rbc-row-content"
                role="row"
              >
                <div
                  class="rbc-row "
                >
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      12
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      13
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      14
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      15
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      16
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      17
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      18
                    </button>
                  </div>
                </div>
                <div
                  class="rbc-row"
                >
                  <div
                    class="rbc-row-segment"
                    style="flex-basis: 42.857142857142854%; max-width: 42.857142857142854%;"
                  >
                     
                  </div>
                  <div
                    class="rbc-row-segment"
                    style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
                  >
                    <div
                      class="rbc-event"
                    >
                      <div
                        class="rbc-event-content"
                        title="Event 2"
                      >
                        Event 2
                      </div>
                    </div>
                  </div>
                  <div
                    class="rbc-row-segment"
                    style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
                  >
                    <div
                      class="rbc-event"
                    >
                      <div
                        class="rbc-event-content"
                        title="Event 1"
                      >
                        Event 1
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rbc-month-row"
              role="rowgroup"
            >
              <div
                class="rbc-row-bg"
              >
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
              </div>
              <div
                class="rbc-row-content"
                role="row"
              >
                <div
                  class="rbc-row "
                >
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      19
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      20
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      21
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      22
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      23
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      24
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      25
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rbc-month-row"
              role="rowgroup"
            >
              <div
                class="rbc-row-bg"
              >
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg"
                />
                <div
                  class="rbc-day-bg rbc-off-range-bg"
                />
              </div>
              <div
                class="rbc-row-content"
                role="row"
              >
                <div
                  class="rbc-row "
                >
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      26
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      27
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      28
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      29
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      30
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      31
                    </button>
                  </div>
                  <div
                    class="rbc-date-cell rbc-off-range"
                    role="cell"
                  >
                    <button
                      class="rbc-button-link"
                      role="cell"
                      type="button"
                    >
                      01
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="css-1bm8fk6"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
